bin/oneoff/backfill_data/user_cap_state_columns

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env ruby
+
+# This script migrates CAP (Child Account Policy) data from properties to separate indexed columns.
+
+require 'ruby-progressbar'
+require_relative '../../../dashboard/config/environment'
+
+# @see https://github.com/code-dot-org/code-dot-org/pull/52244
+CAP_RELEASE_DATE = DateTime.parse('2023-06-09')
+
+users = User.with_deleted.where(current_sign_in_at: CAP_RELEASE_DATE..)
+total_updated = 0
+failed_user_ids = []
+
+update_progress_title = proc {|num_updates = 1| "Processed[%c/%C]: |%W| Updated: #{total_updated += num_updates} %a"}
+progress_bar = ProgressBar.create(total: users.count, format: update_progress_title.call(0))
+
+users.select(
+  :id,
+  :cap_state,
+  :cap_state_date,
+  "properties->>'$.child_account_compliance_state' AS original_cap_state",
+  "properties->>'$.child_account_compliance_state_last_updated' AS original_cap_state_date"
+).find_each(batch_size: 10_000) do |user|
+  next if user.cap_state || user.original_cap_state.blank?
+
+  user.update_columns(
+    cap_state: user.original_cap_state == 's' ? Policies::ChildAccount::ComplianceState::LOCKED_OUT : user.original_cap_state,
+    cap_state_date: user.original_cap_state_date.presence && DateTime.parse(user.original_cap_state_date)
+  )
+
+  progress_bar.format update_progress_title.call
```

Comment:
We will also need to log data about successful updates as well as unsuccessful ones; in the event of any kind of unexpected outcomes, it's critical that we have all the information we need to get our users back into their original state. At a minimum, this means we need to record the ids of all affected users as well as the original data values in the affected columns before the update is applied

Reply:
Do we need to track all affected users individually, since, in this case, we can easily identify them by the statement `cap_state IS NOT NULL`?

Reply:
Yes, we do; we will need to be able to distinguish between users who arrived at that state via this script and users who got there by any other method.

Reply:
I agree we would need it if we were going to use it multiple times. However, in this case, we only need to copy-paste data from one column to another once, and we won't need to touch it again. I would also prefer to delete the script immediately after the data is populated.

Reply:
https://github.com/code-dot-org/code-dot-org/pull/59962/commits/31f5b2c2a232523a12bcc7f7e25ad60925a1cfe3